pkg/util/cloud_prepper.go

```diff
@@ -47,7 +47,8 @@ func (p CloudPrepper) SupportsImage() bool {
 	}
 	strippedSource := strings.Replace(p.ImagePrepper.Source, RemotePrefix, "", -1)
 	_, err := reference.Parse(p.ImagePrepper.Source)
-	if err != nil {
+	if err == nil {
```

Comment:
i would reverse these conditionals
```golang
if err != nil {
  // log the error or inspect it
  return false 
} 

p.ImagePrepper.Source = strippedSource
return true
```

I also don't like that calling a validation function actually mutates my p.ImagePrepper.Source.  Is there a better pattern for prefixing these correctly?  Seems very prone to break (repeated calls, calling this with the with/without a prefix, etc.)

Reply:
Thinking a little more, I think the real problem here is that ImagePrepper shouldn't have `Source` be a `string` but it should actually be a containers/image `types.ImageReference`, which is exactly what you can pass into all of the containers/images functions without have to worry about parsing it at the end.

https://github.com/containers/image/blob/master/types/types.go#L37-L90

Reply:
There are a few problems here, the biggest of which is that the `ImagePrepper` is supposed to be an abstraction inside of the `Prepper` interface, and not all implementations have a `types.ImageReference` (e.g. the `TarPrepper`) doesn't use this. So there isn't a 1:1 conversion of the source to the `ImageReference`.

I do agree that mutating the source in the validation function isn't a great pattern though, I'm working on fixing this. We should probably do all source parsing in the client code, and keep everything constant once we get down into the library code that actually does the processing.